langchain/vectorstores/faiss.py

```diff
@@ -180,13 +180,19 @@ def add_embeddings(
         return self.__add(texts, embeddings, metadatas=metadatas, ids=ids, **kwargs)
 
     def similarity_search_with_score_by_vector(
-        self, embedding: List[float], k: int = 4
+        self,
+        embedding: List[float],
+        k: int = 4,
+        filter: Optional[Dict[str, str]] = None,
+        fetch_k: int = 20,
     ) -> List[Tuple[Document, float]]:
         """Return docs most similar to query.
 
         Args:
             embedding: Embedding vector to look up documents similar to.
             k: Number of Documents to return. Defaults to 4.
+            filter (Optional[Dict[str, str]]): Filter by metadata. Defaults to None.
+            fetch_k: Number of Documents to fetch after filtering
```

Comment:
this should read "before filtering" right?

Reply:
Ohh yes. I will edit and add examples in docs as well. Thanks!

Reply:
Now it is possible to filter after docs are found! Is it possible to filter first by metadata and then look for a match to the request?

Reply:
I'm looking for that too